File: backend/src/Models/Subtask.cs
namespace backend.Models

{
    public class Subtask
    {
        public int Id { get; set; }
        public string Description { get; set; } = string.Empty;
        public bool IsDone { get; set; }
        public int TaskId { get; set; }
        public virtual Task? Task { get; set; }
    }
}

File: frontend/src/utils/requests/Subtask.ts
import axios from "axios";

export interface Subtask {
    Id: number;
    Description: string;
    IsDone: boolean;
    TaskId: number;
}

export const updateSubtask = async (id: number, isDone: boolean, description?: string) : Promise<Subtask> => {
    return axios
        .put(`http://localhost:5002/api/Subtask/${id}`, {
            isDone,
            description
        }, {
            headers: {
                Authorization: `Bearer ${localStorage.getItem("token")}`,
            }
        })
        .then((response) => {
            return response.data;
        });
}